### ISSUE TITLE
Building the C4 dataset in DataFlow from GCP but stuck on create_wet_path_urls and ReadAllFromText

### ISSUE THREAD

**Issue by chrisjihee** [tensorflow/datasets#2964]
**What I need help with / What I was wondering**
What I need help with downloading C4 dataset in DataFlow from GCP.

**What I've tried so far**
I tried the command at https://github.com/google-research/text-to-text-transfer-transformer#c4
And I tried to modify the tensorflow_dataset code for Korean C4 dataset and publish it to https://pypi.org/project/tensorflow-datasets-ko/.
So, my changed command is below.
```
pip install tensorflow-datasets-ko
echo 'tensorflow-datasets-ko' > /tmp/beam_requirements.txt
python -m tensorflow_datasets.scripts.download_and_prepare --datasets=c4/exobrain --data_dir=gs://$MY_BUCKET/tensorflow_datasets --beam_pipeline_options="project=$MY_PROJECT,job_name=c4,staging_location=gs://$MY_BUCKET/binaries,temp_location=gs://$MY_BUCKET/temp,runner=DataflowRunner,requirements_file=/tmp/beam_requirements.txt,experiments=shuffle_mode=service,region=$MY_REGION,autoscaling_algorithm=NONE,num_workers=500"
```

On the Dataflow section in Google Cloud Console,
I found my job, but create_wet_path_urls and ReadAllFromText part was running after about 10 hours.

(On running)
![image](https://user-images.githubusercontent.com/2203944/105124835-5188d300-5b1e-11eb-8b22-df210c77c6a5.png)

**It would be nice if...**
If there is any problem on my trial, please let me know.
Thank you.

**Environment information**
* Operating System: Ubuntu 18.04
* Python version: 3.8.5
* `tensorflow-datasets`/`tfds-nightly` version: tensorflow-datasets-4.2.0

**Comment by Conchylicultor** [comment]
@adarob FYI

**Comment by adarob** [comment]
`create_wet_path_urls` should execute immediately, so there must be a bug in your updated implementation. I'd suggest testing with fewer workers until you get past that point. You could also try running it locally to test.

**Comment by chrisjihee** [comment]
@adarob 
Thank you for your advice.
I found the problem! The problem was modifying directly tensorflow_datasets code.

Firstly I was worried about package name conflict, so I changed the package name and some parts for importing module.
Finally I noticed rather than modifying directly tensorflow_datasets code, adding a new custom dataset and importing on the command line!

My changed code is on https://github.com/chrisjihee/tensorflow-datasets-ko/blob/master/tensorflow_datasets_ko/text/c4ko.py#L212
And my changed command is on https://github.com/chrisjihee/tensorflow-datasets-ko/blob/master/run.sh

Lastly, I am wondering about using multiple workers.
What is the difference these two options?
1. max_num_workers=500
2. autoscaling_algorithm=NONE,num_workers=500
